### ISSUE TITLE
Side-Nav Bug in latest Chrome mobile

### ISSUE THREAD

**Issue by holgermertens** [Dogfalo/materialize#4082]
### Description
While SideNav is opened you can swipe in latest Chrome Mobile in the not-nav-area top or down. 
Then the UI act strange.

### Repro Steps
- open a site with SideNav in latest Chrome (v55) on mobile device (for example http://materializecss.com/navbar.html )
- open SideNav
- scroll on the not side-nav area top or down

### Screenshots / Codepen
https://youtu.be/p54uGC7sO7I

**Comment by tomscholz** [comment]
Can reproduce

**Comment by leftdead7** [comment]
Any updates on this issue?

**Comment by YannGrosjean** [comment]
Any update ? 
I have the problem to :(

**Comment by fega** [comment]
@holgermertens @leftdead7 @YannGrosjean  this is still an issue? I cannot reproduce it in chrome 57.0

**Comment by YannGrosjean** [comment]
Yeah, it's still an issue. 
It's append on chrome mobile on my android phone (version 56.0.2924.87). 
I will try to update to  chrome 57.

**Comment by YannGrosjean** [comment]
I just try on chrome 57 and it's doesn't work better. 
They is two think by the way :
First, the side nav close on scroll and pop randomly after that (only on chrome). 
Second : if I swip to hide the side nav, the overlay stay visible since I click on the screen, but it's also append on firefox mobile.

Any one else have the problem too ? 

If you need some illustration, I can make a video.

**Comment by RaymondBakker** [comment]
Everyone has this issue and is fairly easy to reproduce given you watch the video in the first post.

It seems to be related to receiving incorrect touch coordinates (0, 0) and the navbar script doesn't know how to handle this. It is most likely caused by a change in Chrome.

An option would be to check if the coordinates are (0, 0) but this causes other glitches.

**Comment by holgermertens** [comment]
Of course, it's still there. It's also on latest Opera Mobile.

**Comment by FlatDragon** [comment]
I newbe with js. However, i made temporary solution, which bad i suppose, but anyway.

So, i replaced this string (in sideNav.js):
`$dragTarget = $('<div class="drag-target"></div>').attr('data-sidenav', menuId);`

With this:
`$dragTarget = $('<div class="drag-target" ontouchmove="closeSideNavOnMobileTouch()"></div>').attr('data-sidenav', menuId);`

And write this function:
closeSideNavOnMobileTouch() {            
     $('#sidenav-overlay').click();
 }

So when u tap overlay and tring to scroll page, this function just close side-nav and enable page scrolling again. 
I test it in Yandex.Browser 17 and Chrome 57 - its working.

Sorry for this bad solution, i just hope it will help somebody.

**Comment by FlatDragon** [comment]
Temporary solution. Replace this string:
`$dragTarget = $('<div class="drag-target"></div>').attr('data-sidenav', menuId);`

With this:
`$dragTarget = $('<div class="drag-target" ontouchmove="preventOverlay(event)"></div>').attr('data-sidenav', menuId);`

Add this in the end of file (Sorry, i'm newbe with JS, so don't know how write it correctly):
`function preventOverlay(event) {
    event.preventDefault();            
}`

This working at Chrome 57 and Yandex.Brower 17.

**Comment by rayriffy** [comment]
It's look like you can't scrolling anymore after open sideNav. I also got fix here https://github.com/Dogfalo/materialize/issues/4622. All you need to do is edit ALL materialize.js on "Enable Scrolling" section from "overflow: ' '" to "overflow: 'visible'"

**Comment by redboul** [comment]
Hi,
It seems to comes from [sideNav.js#L189](https://github.com/Dogfalo/materialize/blob/v0.98.2/js/sideNav.js#L189):
```javascript
menu.css('transform', 'translateX(' + (x - options.menuWidth) + 'px)');
```

when swipping from the overlay, `x` equals 0, therefore it translates the menu out but does not close it correctly.
the workaround would be to give `x` a fallback value to `300`. The code would change to:
```javascript
menu.css('transform', 'translateX(' + ((x || 300) - options.menuWidth) + 'px)');
```
**FYI**: it only occurs when the menu is open on the left edge.

**Comment by RaymondBakker** [comment]
@redboul while this is definitely a step in the right direction there are still minor glitches with this workaround.

If you open the menu by swiping, then close the menu by swiping and then press the menu button in the top bar the sidenav will briefly open and close.

**Comment by acburst** [comment]
Fixed in 9cbc2fd
